File: nvpl_blas/c/drotg.c
/******************************************************************************
 * Content: 
 *     This example demonstrates use of API as below:
 *     cblas_drotg
 *
 ******************************************************************************/
#include "example_helper.h"

int main() {
    double a = 2.3f;
    double b = 4.6f;
    double c = 0.0f;
    double s = 0.0f;

    printf("\nExample: cblas_drotg for computing the parameters for a Givens rotation\n");
    printf("\n#### args: a=%f, b=%f, c=%f, s=%f\n",a, b, c, s);

    // call cblas_drotg
    cblas_drotg(&a, &b, &s, &c);

    // print result
    printf("\nThe result of cblas_drotg:\n");
    printf("a=%f, b=%f, c=%f, s=%f\n",a, b, c, s);

    return 0;
}

File: nvpl_blas/f77/drotg.f
!******************************************************************************
! Content:
!     This example demonstrates use of API as below:
!     DROTG
!
!******************************************************************************
      program   DROTG_MAIN
*
      implicit none

      include "nvpl_blas_f77_blas.fi"

      double precision a, b, c, s
      data             a /2.3/, b /4.6/, c /0.0/, s /0.0/


*
*      Executable Statements
*
      print*
      print 99
*
*       Print input data
      print*
      print 100, a, b, c, s
*
*      Call DROTG subroutine
      call DROTG(a, b, c, s)
*
      print*
      print 101
      print 102, a, b, c, s

      stop
 99   format('Example: DROTG for computing the parameters for a given',
     & ' rotation')
 100  format('#### args: a=',f5.3,', b=',f5.3,', c=',f5.3,
     &       ', s=',f5.3)
 101  format('The result of DROTG: ')
 102  format('a=',f5.3,', b=',f5.3,', c=',f5.3,', s=',f5.3)
      end
